mllib-local/src/test/scala/org/apache/spark/ml/linalg/MatricesSuite.scala

```diff
@@ -160,22 +160,385 @@ class MatricesSuite extends SparkMLFunSuite {
     assert(sparseMat.values(2) === 10.0)
   }
 
-  test("toSparse, toDense") {
-    val m = 3
-    val n = 2
-    val values = Array(1.0, 2.0, 4.0, 5.0)
-    val allValues = Array(1.0, 2.0, 0.0, 0.0, 4.0, 5.0)
-    val colPtrs = Array(0, 2, 4)
-    val rowIndices = Array(0, 1, 1, 2)
+  test("dense to dense") {
+    /*
+      dm1 =  4.0 2.0 -8.0
+            -1.0 7.0  4.0
+
+      dm2 = 5.0 -9.0  4.0
+            1.0 -3.0 -8.0
+     */
+    val dm1 = new DenseMatrix(2, 3, Array(4.0, -1.0, 2.0, 7.0, -8.0, 4.0))
+    val dm2 = new DenseMatrix(2, 3, Array(5.0, -9.0, 4.0, 1.0, -3.0, -8.0), isTransposed = true)
+
+    val dm3 = dm1.toDense
+    assert(dm3 === dm1)
+    assert(!dm3.isTransposed)
+    assert(dm3.values.equals(dm1.values))
+
+    val dm4 = dm1.toDenseMatrix(false)
```

Comment:
I would like to make `toDenseMatrix` as private, and we test against `toDenseRowMajor` which is more explicit.